### DISCUSSION TITLE
tutorial yes/no version of the model?

### DISCUSSION THREAD

**Discussion by subzero882** [coqui-ai/STT#2020] [Q&A]
Is there a tutorial on how to retrain coqui into the yes/no only model?

I'm building some software looking to use command words, and i'm having some issues in understanding how to take STT as is and have it reduce to a subset of words. like how the yes/no model would work

**Comment by JRMeyer** [comment]
This is a great request... there actually isn't one (yet!).

There are similar, existing tutorials here: https://github.com/coqui-ai/STT/tree/main/notebooks

when we've got a yes/no tutorial I'll post here:)

**Comment by JRMeyer** [comment]
Hi @subzero882 -- here's a quick tutorial on training STT models for yes/no... let me know what you think!

https://colab.research.google.com/github/JRMeyer/STT/blob/jupyter/notebooks/yesno_tutorial.ipynb

**Reply by subzero882** [reply]
@JRMeyer First, I want to say how amazed i am at the speed of your response!

Second, I didn't realize that the notebooks folder had tutorials in it, i was just going by the ReadTheDoc page. So that is very helpful.

Third, after looking at the notebook I think the tutorial that helps the most is this one, https://colab.research.google.com/github/coqui-ai/STT/blob/main/notebooks/easy_transfer_learning.ipynb but going over the Yes/No one was helpful as well.

Finally, I am using the docker image for my training which i think i have all setup now. Based on the tutorials you posted (not well versed in colab though I think I should learn it), and after looking over the tutorials above I think I have a direction way better than I did before so thank you very much for that. I think it would also help to see some tutorials similar to the ones you have cover doing this with the docker image.

With what I am trying to do (command words), does that sound right to use the transfer learning for a checkpoint in the huge-vocab, and then just have the command words as data for that? Also, since i would be using transfer learning, does it sound reasonable to only have each command word recorded only a couple of time for a few people, or do you think I'll need hours of data for that?


Thank you very much!!!!

**Reply by JRMeyer** [reply]
>With what I am trying to do (command words), does that sound right to use the transfer learning for a checkpoint in the huge-vocab, and then just have the command words as data for that? Also, since i would be using transfer learning, does it sound reasonable to only have each command word recorded only a couple of time for a few people, or do you think I'll need hours of data for that?

the best approach for _you_ and _your use case_ will depend on a few things, but the fastest and easiest one I would first try out is to just take the `v1.0` model (if you're using English) and use [`hot word boosting`](https://stt.readthedocs.io/en/latest/HotWordBoosting-Examples.html). this way you don't have to fine-tune or re-train anything yourself.

You should come chat with us in our community STT channel:) that's where you will get even faster feedback!

https://gitter.im/coqui-ai/STT

**Reply by subzero882** [reply]
@JRMeyer Thank you for link to the chat.

So previously when we were using deepspeech just to test this all out, we used hotwords but that didn't help completely as we are using some uncommon words and word pairs for commands that probably aren't in the current training data. Example words, 2D map, 3D map, magnify, Heads-up display, etc. 

I will however, try hotwords with the new model as well and see what results I get before I re-train.

**Comment by JRMeyer** [comment]
@subzero882 the model was trained without digits like [0,1,2... ], but rather words [zero, one, two... ], so, you should try spelling out the hotwords like "two d map" and "three d map", and there's no dash, so "heads up display"

**Reply by subzero882** [reply]
Sorry I wasn't clear there, yes I do have the numbers spelled out as well as no dash in the hotwords.

**Reply by JRMeyer** [reply]
If hotwords isn't sufficient, and you have a command-and-control application (which sounds like you might have), then training a custom language model will be your next step to improve performance. 

https://stt.readthedocs.io/en/latest/LANGUAGE_MODEL.html

**Reply by subzero882** [reply]
Ok ya that's what i was thinking, so with that in mind do you think Fine Tuning would be sufficient?

https://stt.readthedocs.io/en/latest/TRANSFER_LEARNING.html
